packages/hoppscotch-backend/src/shared-request/shared-request.service.ts

```diff
@@ -0,0 +1,217 @@
+import { Injectable, OnModuleInit } from '@nestjs/common';
+import { PrismaService } from 'src/prisma/prisma.service';
+import { PubSubService } from 'src/pubsub/pubsub.service';
+import { Shortcode as DBSharedRequest } from '@prisma/client';
+import * as E from 'fp-ts/Either';
+import * as TO from 'fp-ts/TaskOption';
+import * as T from 'fp-ts/Task';
+import { SharedRequest } from './shared-requests.model';
+import {
+  SHARED_REQUEST_INVALID_PROPERTIES_JSON,
+  SHARED_REQUEST_INVALID_REQUEST_JSON,
+  SHARED_REQUEST_NOT_FOUND,
+} from 'src/errors';
+import { stringToJson } from 'src/utils';
+import { AuthUser } from 'src/types/AuthUser';
+import { PaginationArgs } from 'src/types/input-types.args';
+import { UserDataHandler } from 'src/user/user.data.handler';
+import { UserService } from 'src/user/user.service';
+
+const SHORT_CODE_LENGTH = 12;
+const SHORT_CODE_CHARS =
+  'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789';
+@Injectable()
+export class SharedRequestService implements UserDataHandler, OnModuleInit {
+  constructor(
+    private readonly prisma: PrismaService,
+    private readonly pubsub: PubSubService,
+    private readonly userService: UserService,
+  ) {}
+
+  onModuleInit() {
+    this.userService.registerUserDataHandler(this);
+  }
+
+  canAllowUserDeletion(user: AuthUser): TO.TaskOption<string> {
+    return TO.none;
+  }
+
+  onUserDelete(user: AuthUser): T.Task<void> {
+    return async () => {
+      await this.deleteUserSharedRequests(user.uid);
+    };
+  }
+
+  /**
+   * Delete all the Users SharedRequests
+   * @param uid User Uid
+   * @returns number of all deleted user SharedRequests
+   */
+  async deleteUserSharedRequests(uid: string) {
+    const deletedShortCodes = await this.prisma.shortcode.deleteMany({
+      where: {
+        creatorUid: uid,
+      },
+    });
+
+    return deletedShortCodes.count;
+  }
+
+  /**
+   * Converts a Prisma SharedRequest type into the SharedRequest model
+   *
+   * @param sharedRequestInfo Prisma SharedRequest type
+   * @returns GQL SharedRequest
+   */
+  private cast(sharedRequestInfo: DBSharedRequest): SharedRequest {
+    return <SharedRequest>{
+      id: sharedRequestInfo.id,
+      request: JSON.stringify(sharedRequestInfo.request),
+      properties: JSON.stringify(sharedRequestInfo.properties),
```

Comment:
If `properties` is `null`, then we should avoid to `JSON.stringify` it.

<img width="1118" alt="Screenshot 2023-10-27 at 6 10 18 AM" src="https://github.com/hoppscotch/hoppscotch/assets/30154643/896314cb-bebb-471a-8dd5-31dd1c823e1f">